### PR TITLE
Support HY-MT evaluation workflows

### PR DESCRIPTION
**Pull request by hanbitmyths**
## Summary
- Update lm-eval integration to preserve recipe bootstrap settings.
- Improve ORT GenAI evaluation provider handling and runtime GenAI search options.
- Keep model-load-only Hugging Face kwargs out of generation config loading.
- Add accelerator normalization coverage for WebGPU export-only and runtime-required paths.

## Validation
- Ran py_compile for olive/evaluator/lmeval_ort.py, olive/evaluator/olive_evaluator.py, olive/model/handler/mixin/hf.py, and test/hardware/test_accelerator.py.
- Ran git diff --check for the touched Olive files.
- VS Code diagnostics reported no errors for touched accelerator/evaluator files.
- Targeted pytest was attempted but blocked by missing pydantic in this environment.

### CODE REVIEW

File: olive/evaluator/lmeval_ort.py [hunk 1]
```diff
@@ -568,10 +587,16 @@ def __init__(
...
+            self._past_present_share_buffer = genai_config["search"].get("past_present_share_buffer", False)
         self.params = og.GeneratorParams(self.model)
-        self.params.set_search_options(max_length=self.max_length, past_present_share_buffer=False)
+        self.params.set_search_options(
+            max_length=self.max_length,
+            past_present_share_buffer=self._past_present_share_buffer,
+        )
 
-        self.device = device
+        self._device = device
```
[the remark this answers is not in the document]
Comment (hanbitmyths) [reply olive/evaluator/lmeval_ort.py:600]: Fixed in cd03fff4 by updating both logits conversion call sites in model_call() to use self._device. Validated with py_compile, git diff --check, and VS Code diagnostics.

File: olive/evaluator/lmeval_ort.py [hunk 2]
```diff
@@ -515,6 +515,23 @@
 class LMEvalORTGenAIEvaluator(LMEvalOnnxBase):
     """Evaluate a model using ONNX Runtime GenAI."""
 
+    _ORT_GENAI_PROVIDER_NAMES = {
```
[the remark this answers is not in the document]
Comment (hanbitmyths) [reply olive/evaluator/lmeval_ort.py]: Fixed in cd03fff4 by annotating _ORT_GENAI_PROVIDER_NAMES as ClassVar[dict[str, str]]. Validated with /usr/bin/ruff check olive/evaluator/lmeval_ort.py --select RUF012.

**Review by xiaoyu-work — APPROVED** [review]
